### PR TITLE
Hop to right event loop when using transactions

### PR DESCRIPTION
**Pull request by kacperk**
Using transaction cause vapor crash with message `database driver output was not on eventloop`

From what I've understood from MongoKitten code, creating transaction gets new event loop from the pool, but driver code doesn't handle that.

### CODE REVIEW

File: Sources/FluentMongoDriver/MongoDB+Transaction.swift [hunk 1]
```diff
@@ -21,7 +22,7 @@ extension FluentMongoDatabase {
                 return transactionDatabase.abort().flatMapThrowing { _ in
                     throw error
                 }
-            }
+            }.hop(to: context.eventLoop)
```
Comment (siemensikkema) [inline Sources/FluentMongoDriver/MongoDB+Transaction.swift:25]: @kacperk Wouldn't this be the only thing that's needed?

Comment (kacperk) [reply Sources/FluentMongoDriver/MongoDB+Transaction.swift:25]: @siemensikkema from what I was testing it is not enough - it crashes on first query inside of transaction. Without new context for transaction, database object will have different event loop on raw database object and different in context.

File: Sources/FluentMongoDriver/MongoDB+Transaction.swift [hunk 2]
```diff
@@ -9,10 +9,11 @@ extension FluentMongoDatabase {
         }
         do {
             let transactionDatabase = try raw.startTransaction(autoCommitChanges: false)
+            let context =  DatabaseContext.init(configuration: self.context.configuration, logger: self.context.logger, eventLoop: transactionDatabase.eventLoop)
```
Comment (siemensikkema) [inline Sources/FluentMongoDriver/MongoDB+Transaction.swift:12]:
```suggestion
            let context = DatabaseContext(configuration: self.context.configuration, logger: self.context.logger, eventLoop: transactionDatabase.eventLoop)
```

File: Sources/FluentMongoDriver/MongoDB+Transaction.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/FluentMongoDriver/MongoDB+Transaction.swift:25]: Fair enough 🙂 Would you be able to write a test for this fix?

**Comment by siemensikkema** [comment]
@Joannis does this fix seem like a good idea to you?

**Comment by Joannis** [comment]
Morning everyone! Since the MongoDatabase should already have registered the preferred callback EventLoop, this is a bug in MongoKitten. It's a really simple fix, and I strongly believe this is not to be fixed in the Fluent driver.

**Comment by siemensikkema** [comment]
@Joannis I'm glad I asked 🙂 

@kacperk maybe you could hop (pun intended) over to https://github.com/OpenKitten/MongoKitten and raise an issue there (or better, a PR!).

**Comment by Joannis** [comment]
Before writing the comment here I've already created and tested a fix.

**Comment by siemensikkema** [comment]
Fixed in https://github.com/OpenKitten/MongoKitten/releases/tag/6.6.5

**Comment by Joannis** [comment]
@siemensikkema would it be sensible to add a test for this anyways?

**Comment by siemensikkema** [comment]
@Joannis not sure, that seems to me like it would be testing dependencies (MongoKitten in this case). The existing fluent-mongo-driver implementation is sort of the obvious one IMO which assumes that MongoKitten behaves according to expectations. In other words, if there should be a test, it should be on MongoKitten, no?

File: Sources/FluentMongoDriver/MongoDB+Transaction.swift (hunk 1 above)
Comment (kacperk) [reply Sources/FluentMongoDriver/MongoDB+Transaction.swift:25]: @siemensikkema is there any documentation telling how to setup test environment?

**Pull request closed without merging**